service/src/main/java/bio/terra/landingzone/job/LandingZoneJobService.java

```diff
@@ -384,6 +385,18 @@ public void verifyUserAccess(BearerToken bearerToken, String jobId) {
       if (landingZoneId == null) {
         throw new JobNotFoundException("The landing zone does not exist for job");
       }
+      var azureLandingZoneRequest =
+          inputParameters.get(
+              LandingZoneFlightMapKeys.LANDING_ZONE_CREATE_PARAMS, LandingZoneRequest.class);
+      // Check that the calling user has "link" permission on the billing profile resource in Sam
+      SamRethrow.onInterrupted(
```

Comment:
Why do we need the `billing-profile` check here? Seems like the `landing-zone` check should be sufficient, no?

Reply:
(It's just for the delete case, that we can't rely on the `landing-zone` resource existing)

Reply:
This is for the case when create fails, the undo step deletes the sam resource and the job can't be retrieved.